tests/test_hub.py

```diff
@@ -0,0 +1,185 @@
+                        "split": "test",
+                        "revision": "abc",
+                        "args": {"a": 1, "b": 2},
+                    },
+                    "metrics": [
+                        {
+                            "type": "metric",
+                            "value": 1.0,
+                            "name": "dummy_metric",
+                            "config": "default",
+                            "args": {"hello": 1, "world": 2},
+                        },
+                    ],
+                }
+            ]
+        }
+    ]
+}
+
+
+@patch("evaluate.hub.get_allowed_tasks", lambda x: ["dummy-task"])
+@patch("evaluate.hub.dataset_info", lambda x: True)
+@patch("evaluate.hub.model_info", lambda x: True)
+@patch("evaluate.hub.metadata_update")
+@patch("evaluate.hub.known_task_ids")
+class TestHub(TestCase):
+    @pytest.fixture(autouse=True)
+    def inject_fixtures(self, caplog):
+        self._caplog = caplog
+
+    def setUp(self):
+        self.metric = DummyMetric()
+        self.metric.add()
+        self.args = {"hello": 1, "world": 2}
+        self.result = self.metric.compute()
+
+    def test_push_metric_required_arguments(self, known_task_ids, metadata_update):
+        push_to_hub(
+            repo_id="username/repo",
+            metric_value=self.result["accuracy"],
+            metric_name=self.metric.name,
+            metric_type=self.metric.module_type,
+            dataset_name="dataset_name",
+            dataset_type="dataset_type",
+            task_type="dummy-task",
+        )
+
+        metadata_update.assert_called_once_with(repo_id="username/repo", metadata=minimum_metadata, overwrite=False)
+
+    def test_push_metric_missing_arguments(self, known_task_ids, metadata_update):
+        with pytest.raises(TypeError):
+            push_to_hub(
+                repo_id="username/repo",
+                metric_value=self.result["accuracy"],
+                metric_name=self.metric.name,
+                metric_type=self.metric.module_type,
+                dataset_name="dataset_name",
+                dataset_type="dummy-task",
+            )
+
+    def test_push_metric_invalid_arguments(self, known_task_ids, metadata_update):
+        with pytest.raises(TypeError):
+            push_to_hub(
+                repo_id="username/repo",
+                metric_value=self.result["accuracy"],
+                metric_name=self.metric.name,
+                metric_type=self.metric.module_type,
+                dataset_name="dataset_name",
+                dataset_type="dataset_type",
+                task_type="dummy-task",
+                random_value="incorrect",
+            )
+
+    def test_push_metric_extra_arguments(self, known_task_ids, metadata_update):
+        push_to_hub(
+            repo_id="username/repo",
+            metric_value=self.result["accuracy"],
+            metric_name=self.metric.name,
+            metric_type=self.metric.module_type,
```

Comment:
just noticed this now: `module_type` is not the `metric_type`. in evaluate `module_type` can be one of `"metric"`, `"measurement"`, and `"comparison"`. For the `push_to_hub` we only do it for `metrics` anyway . 

```suggestion
            metric_name="Pretty Metric Name",
            metric_type=self.metric.name,
```

It doesn't really matter for the test but just to be consistent. Could you change this for all?